Stub out contact_properties api calls in tests

MattMSumner opened the pull request:
Why:

These tests can fail in CI and depend on an API key being available.

This PR:

Removes the need for an API key by using oauth but also mocks out all
the api requests so it can run consistently in CI.

This also updates the endpoint to be accurate. The previous value currently redirects to the new value.

SViccari commented on lib/contact_property.js:
```diff
@@ -6,7 +6,7 @@ class Properties {
     return this.client._request({
       method: 'GET',
-      path: '/contacts/v2/properties',
+      path: '/properties/v1/contacts/properties',
```
nice 👍

SViccari commented on test/contact_properties.js:
```diff
@@ -1,164 +1,296 @@
+  const hubspot = new Hubspot({
+    accessToken: process.env.ACCESS_TOKEN || 'fake-token',
+  })
+  const properties = {
```
Are these properties being used to create a Contact Property Group and a Contact Property? That seems a bit odd to me as they're not composed of the same fields. Fetching a Contact Property Group returns:
```
{
    "name": "colors2",
    "displayName": "Details about color choices",
    "displayOrder": 81690,
    "hubspotDefined": false
}
```

Fetching a Contact Property returns:
```
{
    "name": "account_type",
    "label": "account_type",
    "description": "",
    "groupName": "contactinformation",
    "type": "enumeration",
    "fieldType": "select",
    "options": []
}
```

SViccari commented on test/contact_properties.js:
```diff
@@ -1,164 +1,296 @@
+    hubspot.contacts.properties.createGroup(properties)
+
+  const deleteTestProperty = name => hubspot.contacts.properties.delete(name)
+  const createTestProperty = properties =>
```
small naming suggestion: `createTestContactProperty`. Just to clarify this is a `ContactProperty` vs a `CompanyProperty`.

SViccari commented on test/contact_properties.js:
```diff
@@ -1,164 +1,296 @@
+    const description = 'Updated display name'
+    const updateEndpoint = {
+      path: `/properties/v1/contacts/properties/named/${properties.name}`,
+      response: { ...properties, description },
```
I'm curious about the use of the `...` syntax. Is this returning the properties hash as an array?

MattMSumner replied:
It's spatting the object into this object. Checkout https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#Spread_in_object_literals

SViccari commented on test/contact_properties.js:
```diff
@@ -1,164 +1,296 @@
+      const description = 'Updated display name'
+      const createEndpoint = {
+        path: '/properties/v1/contacts/properties',
+        statusCode: 409,
```
> The HTTP 409 Conflict response status code indicates a request conflict with the current state of the server.

TIL 🌈

MattMSumner replied:
I think I would prefer to remove this functionality but since we already have a big breaking change I didn't want to introduce another. This upsert surprised me.

SViccari commented on test/contact_properties.js:
```diff
@@ -1,164 +1,296 @@
+
+      it('returns a 200', function() {
+        return hubspot.contacts.properties
+          .createGroup({ name, displayName: 'Test Group' })
```
I believe a `displayName` is not required to create a Contact Property Group. There's no harm in leaving it in but you could also remove it so the test focuses on only the fields that are required.

SViccari reviewed (approved):
A few small comments but overall, LGTM 🐳

MattMSumner committed:
Stub out contact_properties api calls in tests

Why:

These tests can fail in CI and depend on an API key being available.

This PR:

Removes the need for an API key by using oauth but also mocks out all
the api requests so it can run consistently in CI.
